pkg/router/controller/factory/factory.go

```diff
@@ -60,11 +62,17 @@ func (factory *RouterControllerFactory) Create(plugin router.Plugin) *controller
 
 	endpointsEventQueue := oscache.NewEventQueue(cache.MetaNamespaceKeyFunc)
 	cache.NewReflector(&endpointsLW{
-		client:    factory.KClient,
+		client:    factory.EndpointsNamespacer,
 		namespace: factory.Namespace,
 		// we do not scope endpoints by labels or fields because the route labels != endpoints labels
 	}, &kapi.Endpoints{}, endpointsEventQueue, factory.ResyncInterval).Run()
 
+	svcStore := cache.NewStore(cache.MetaNamespaceKeyFunc)
```

Comment:
You have to handle permission denied here - routers will be upgraded and may not have access to services until an admin upgrades their permissions.  So you need to test against a server without permission and ensure that the cache is loaded, warns the admin (via a log message) and then works correctly.